app/src/main/java/org/p2p/wallet/history/interactor/mapper/SolanaInstructionParser.kt

```diff
@@ -1,125 +1,90 @@
-                        .onFailure(onErrorLogger)
-                        .getOrNull()
-                )
-            }
-            else -> {
-                parseTransaction(transactionRoot, signature, findMintAddress)
-            }
-        }
-    }
+        val transactionInstructions = transactionRoot.transaction?.message?.instructions ?: emptyList()
 
-    private fun parseTransaction(
-        transactionRoot: ConfirmedTransactionRootResponse,
-        signature: String,
-        findMintAddress: (String) -> String
-    ): List<TransactionDetails> {
-        return transactionRoot.transaction?.message
-            ?.instructions
-            ?.mapNotNull { parseInstructionByType(signature, transactionRoot, it, findMintAddress) }
-            ?.toMutableList()
-            .orEmpty()
-    }
+        val items = transactionInstructions.map { instruction ->
 
-    private fun parseInstructionByType(
-        signature: String,
-        transactionRoot: ConfirmedTransactionRootResponse,
-        parsedInstruction: InstructionResponse,
-        findMintAddress: (String) -> String
-    ): TransactionDetails? {
-        val parsedInfo = parsedInstruction.parsed
-        return when (parsedInfo?.type) {
-            "burnChecked" -> {
-                parseBurnOrMintTransaction(
-                    signature = signature,
-                    transactionRoot = transactionRoot,
-                    parsedInfo = parsedInfo
-                )
-            }
-            "transfer", "transferChecked" -> {
-                parseTransferTransaction(
-                    signature = signature,
-                    transactionRoot = transactionRoot,
-                    parsedInfo = parsedInfo,
-                    findMintAddress = findMintAddress
-                )
-            }
-            "closeAccount" -> {
-                parseCloseTransaction(
-                    parsedInstruction = parsedInstruction,
-                    parsedInfo = parsedInfo,
-                    transactionRoot = transactionRoot,
-                    signature = signature,
-                )
-            }
-            "create" -> {
-                parseCreateAccountTransaction(
-                    signature = signature,
-                    transactionRoot = transactionRoot
-                )
-            }
-            else -> {
-                parseUnknownTransaction(
-                    signature = signature,
-                    transactionRoot = transactionRoot
-                )
+            val parsedInfo = instruction.parsed
+            val type = parsedInfo?.type
+            when (TransactionDetailsType.valueOf(type)) {
+                TransactionDetailsType.TRANSFER, TransactionDetailsType.TRANSFER_CHECKED -> {
+                    parseTransferTransaction(
+                        signature = signature,
+                        transactionRoot = transactionRoot,
+                        parsedInfo = parsedInfo!!,
```

Comment:
parsedInfo!! not safe here
we need to make it not null or check it before or make some fallback info if it will be null